tasks_file_string/tasks_string.py

```diff
@@ -0,0 +1,193 @@
+            if check_string_len(new_unit_string):  # checking the length of the string according to the condition specified in the task
+                remove_words_list.append(new_unit_string)
+        elif check_string_len(unit_string):
+            remove_words_list.append(unit_string)
+    if not remove_words_list or len(remove_words_list) == 1:
+        return False
+    if not len(remove_words_list) % 2 == 0:
+        remove_words_list.pop(-1)
+    return remove_words_list
+
+
+def remove_words_by_condition(source_file_name="task_1_text.txt"):
+    """Main function which starts the process of checking and transforming the transferred file"""
+    source_data = open_file(source_file_name)
+    result_file_path = check_result_file(source_file_name)
+    for one_line in source_data:
+        remove_words = check_one_line(one_line)
+        if not remove_words:
+            write_file(result_file_path, one_line)
+            continue
+        for remove_word in remove_words:
+            find_index = one_line.find(remove_word)
+            first_part_new_string = one_line[:find_index]
+            second_part_new_string = one_line[(find_index + len(remove_word)):]
+            one_line = first_part_new_string + second_part_new_string
+        write_file(result_file_path, one_line)
+    return f"Result file:{result_file_path}"
+
+
+def check_string_len(string):  # moved to a single function as it is used several times
+    """Сhecking a given condition in a task"""
+    if 3 <= len(string) <= 5:
+        return True
+    return False
+
+
+# 2)Текстовый файл содержит записи о телефонах и их владельцах. Переписать в другой файл телефоны
+# тех владельцев, фамилии которых начинаются с букв К и С.
+
+def sorted_phone_book(source_file_name="task_2_phone_book.txt"):
+    """Function checking phone book and filter by given condition after writing the filtered data to file"""
+    sorted_keys = ("к", "с")
+    source_data = open_file(source_file_name)
+    result_file_path_write = check_result_file(source_file_name)
+    for unit_phone_note in source_data:
+        prepared_unit_phone_note = unit_phone_note.lower()
+        for sorted_key in sorted_keys:
+            if prepared_unit_phone_note.find(sorted_key) == 0:
+                write_file(result_file_path_write, unit_phone_note)
+    return f"Result file:{result_file_path_write}"
+
+
+# 3) Получить файл, в котором текст выровнен по правому краю путем равномерного добавления пробелов.
+
+def right_align(source_file_name="task_3_right_align.txt", fillchar=" "):
+    """Function right-aligns text fill blank space with given fill char"""
+    source_file_data = open_file(source_file_name)
+    result_file_path = check_result_file(source_file_name)
+    max_line_length_in_file = max(len(line) for line in source_file_data) # determining the maximum line width
+    for line in source_file_data:
+        wight = (max_line_length_in_file - len(line)) + len(line)
```

Comment:
Seems like this var is extra, 'max_line_length_in_file' can be used

Reply:
You are right